### ISSUE TITLE
Proper error handling

### ISSUE THREAD

**Issue by elifoster** [FTB-Gamepedia/MediaWiki-Butt-Ruby#32]
Right now, if there's an error, we simply return the error code as a String. This makes it difficult or weird to check for errors. For example, we want to be able to do this, as is idiomatic in Ruby:
```ruby
begin
  hash = butt.some_function
  return hash['important_value']
rescue SomeError
  p 'Oh no we errored'
end
```

but right now, we have to do this:
```ruby
hash = butt.some_function
if hash.is_a?(String)
  p 'Oh no we errored'
else
  return hash['important_value']
end
```

or this, for methods that return a String either way
```ruby
string = butt.other_function
if /Error:/ =~ string
  p 'Oh no we errored'
else
  return string.reverse!
end
```

This is awful.

**Comment by elifoster** [comment]
(See #42).

We have a couple options I've thought of: 

Categorized errors with a description given in the constructor:
```ruby
# We have determined that the edit has errored:
fail EditError.new(error_message)
```

Specifically defined errors
```ruby
# We have determined that the edit has errored
case error_message
when 'badtoken'
  fail BadToken.new
# ...
end
```

The benefits to the first: 
* No chance for unidentified errors, for example, when an extension adds its own errors. There's no way we could add a specific error class for every possible error in every extension.

The benefits to the second:
* Performing different things based on different kinds of errors is much easier. Instead of rescuing a single error and checking the error message, the various errors could be rescued individually, thus making the code more maintainable.

@APerson241 @xbony2

**Comment by APerson241** [comment]
(For the record, I strongly support the first one, for the reason you mentioned.)

**Comment by APerson241** [comment]
@elifoster and @xbony2, what are your opinions on option 1 vs option 2?

**Comment by xbony2** [comment]
Whatever work.

**Comment by elifoster** [comment]
I like option 1

**Comment by APerson241** [comment]
I've implemented option 1 (for just `edit()`) in #42.

**Comment by elifoster** [comment]
I have finally finished this up.
